frontend/components/PageOverlay/index.jsx

```diff
@@ -0,0 +1,102 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import {
+  HtmlSanitizer,
+  Link,
+  Button,
+  Modal,
+  Grid,
+} from '@shopgate/engage/components';
+import appConfig from '@shopgate/pwa-common/helpers/config';
+import styles from './style';
+
+/**
+ * PageOverlay Component
+ *
+ * This component displays an overlay page with a headline, HTML content,
+ * and app store buttons for iOS and Android devices.
+ *
+ * @param {Object} props.buttonStoreText - Contains button texts for iOS and Android
+ * @param {boolean} props.showShopLogo - Indicates whether to show the shop logo
+ * @param {string} props.headline - The main headline of the overlay page
+ * @param {string} props.htmlContent - The HTML content displayed within the page
+ * @param {boolean} props.isIos - Indicates whether the current device is an iOS device
+ * @param {Object} props.storeLinks - Contains store links for iOS and Android
+ *
+ * @returns {JSX.Element}
+ */
+const PageOverlay = ({
+  buttonStoreText,
+  headline,
+  htmlContent,
+  isIos,
+  shopLogoUrl,
+  showShopLogo,
+  storeLinks,
+}) => {
+  const storeLink = isIos ? storeLinks.ios : storeLinks.android;
+  const buttonLabel = isIos ? buttonStoreText.ios : buttonStoreText.android;
+
+  return (
+    <Modal
+      classes={{
+        content: styles.modalContent,
+        layout: styles.modalLayout,
```

Comment:
to guarantee that this overlay is shown "above" the cookie-consent and push opt-in modals please add a `container` property and assign a class with a z-index > 2000